src/d/actor/d_a_shand.cpp

```diff
@@ -4,91 +4,488 @@
+            if(actor->field_30C != NULL){
+                actor->mState = 1;
+                actor->field_2BA = 0;
+                actor->field_2BC[0] = l_HIO.u8;
+                actor->field_2BC[1] = l_HIO.u8 + 90 + REG0_S(2);
+                actor->field_2F8 = 3.0f;
+                actor->field_2FC = 40.0f;
+                actor->field_300 = cM_rndF(20.0f) + 30.0f;
+                *actor->field_314 = 1;
+                cXyz particle_scale(0.5f, 0.5f, 0.5f);
+                dComIfGp_particle_set(19, &hit_atInfo.mpActor->eyePos, NULL, &particle_scale);
+                dComIfGp_particle_set(22, &hit_atInfo.mpActor->eyePos, NULL, &particle_scale);
+            }
+        }
+    }
 }
 
 /* 000021EC-0000225C       .text daShand_Execute__FP11shand_class */
-static BOOL daShand_Execute(shand_class*) {
-    /* Nonmatching */
+static BOOL daShand_Execute(shand_class* actor) {
+    actor->mExecuteCount++;
+    
+    
+    for(int i = 4, j = 0; i != 0; i--, j++){
+        if(actor->field_2BC[j] != 0)
+            actor->field_2BC[j]--;    
+    }
+
+    if(actor->field_2C4 != 0)
+        actor->field_2C4--;
+
+    hand_move(actor);
+
+    return TRUE;
 }
 
 /* 0000225C-00002264       .text daShand_IsDelete__FP11shand_class */
 static BOOL daShand_IsDelete(shand_class*) {
-    /* Nonmatching */
+    return TRUE;
 }
 
 /* 00002264-000022D4       .text daShand_Delete__FP11shand_class */
-static BOOL daShand_Delete(shand_class*) {
-    /* Nonmatching */
+static BOOL daShand_Delete(shand_class* actor) {
+    dComIfG_resDelete(&actor->mPhs, "Shand");
+    if(actor->mHasHIO){
+        hio_set = false;
+        mDoHIO_deleteChild(l_HIO.mId);
+    }
+
+    return TRUE;
 }
 
 /* 000022D4-00002360       .text useHeapInit__FP11shand_class */
-void useHeapInit(shand_class*) {
-    /* Nonmatching */
+static BOOL useHeapInit(shand_class* actor) {
+    const int res_index = ((fopAcM_GetParam(actor) & 0xff) == 53) ? 4 : 3;
+    ResTIMG* res = static_cast<ResTIMG*>(dComIfG_getObjectRes("Shand", res_index));
+    
+    if(actor->mLineMat.init(1, 20, res, 1) == FALSE){
+        return FALSE;
+    }
+    else{
+        return TRUE;
+    }
 }
 
 /* 00002360-00002380       .text daShand_solidHeapCB__FP10fopAc_ac_c */
-static BOOL daShand_solidHeapCB(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL daShand_solidHeapCB(fopAc_ac_c* actor) {
+    return useHeapInit(static_cast<shand_class*>(actor));
 }
 
 /* 00002380-00002630       .text daShand_Create__FP10fopAc_ac_c */
-static cPhs_State daShand_Create(fopAc_ac_c*) {
-    /* Nonmatching */
+static cPhs_State daShand_Create(fopAc_ac_c* f_this) {
+    static dCcD_SrcCyl tg_cyl_src = {
+        0,
+        0, 
+        0,
+        0,
+        AT_TYPE_ALL - (AT_TYPE_WATER | AT_TYPE_UNK20000 | AT_TYPE_WIND | AT_TYPE_UNK400000 | AT_TYPE_LIGHT), // 0xff1dfeff 
+        cCcD_TgSPrm_Set_e | cCcD_TgSPrm_IsEnemy_e,
+        cCcD_CoSPrm_Set_e | cCcD_CoSPrm_IsPlayer_e | cCcD_CoSPrm_VsGrpAll_e, 
```

Comment:
You can make all these collision src fields more readable by putting comments with their field names to the left of them, you can just copy paste the comments from another cyl_src/sph_src elsewhere in the decomp